src/Miunie.ConsoleApp/ConsoleMenu.cs

```diff
@@ -0,0 +1,95 @@
+﻿// This file is part of Miunie.
+//
+//  Miunie is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  Miunie is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with Miunie. If not, see <https://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Miunie.ConsoleApp
+{
+    internal class ConsoleMenu<T>
+    {
+        private readonly IEnumerable<T> _items;
+
+        private readonly Func<T, string> _formatter;
+
+        private string _title;
+
+        private int _selectedIndex;
+
+        internal ConsoleMenu(IEnumerable<T> items, Func<T, string> formatter)
+        {
+            _items = items;
+            _formatter = formatter;
+        }
+
+        internal void SetTitle(string title)
+        {
+            _title = title;
+        }
+
+        internal T Present()
+        {
+            if (_items.Count() == 1)
+            {
+                return _items.First();
+            }
+
+            _selectedIndex = 0;
+            ConsoleKeyInfo keyPressed;
+
+            do
+            {
+                PrintTitle();
+                PrintItems(_selectedIndex);
+
+                keyPressed = Console.ReadKey(true);
+                HandleMovementKey(keyPressed);
+
+                Console.Clear();
+            }
+            while (keyPressed.Key != ConsoleKey.Enter);
+
+            return _items.ElementAt(_selectedIndex);
+        }
+
+        private void HandleMovementKey(ConsoleKeyInfo keyPressed)
+        {
+            if (keyPressed.Key == ConsoleKey.DownArrow)
+            {
+                if (++_selectedIndex == _items.Count()) { _selectedIndex = 0; }
+            }
+            else if (keyPressed.Key == ConsoleKey.UpArrow)
+            {
+                if (--_selectedIndex == -1) { _selectedIndex = _items.Count() - 1; }
+            }
+        }
+
+        private void PrintItems(int hoveringNumber)
```

Comment:
I think this method can just use the `_selectedIndex` private field and not get anything passed into it. 🙂 

That should be an easy fix. 😉